File: Medium/1291. Sequential Digits/solution.cs
public class Solution {
    public IList<int> SequentialDigits(int low, int high) {
        var result = new List<int>();
        int num = 0;

        Backtrack(num, low, high, result);
        result.Sort();

        return result;
    }

    private void Backtrack(int num, int low, int high, List<int> result){
        if (num > high) return;
        if (low <= num && num <= high){
            result.Add(num);
        }

        if (num == 0){
            for (int i = 1; i <= 9; i++){
                Backtrack(i, low, high, result);
            }
        }
        else{
            int pre = num % 10;
            if (pre < 9){
                num = num * 10 + (pre + 1);
                Backtrack(num, low, high, result);
            }
        }
        return;
    }
}

File: Medium/1291. Sequential Digits/solution.cpp
class Solution {
public:
    
    void backtrack(long long &num, int &low, int &high, vector<int> &result){
        if (num > high) return;
        
        if (low <= num && num <= high){
            result.push_back(num);
        }
        
        if (num == 0){
            for (int i = 1; i <= 9; i++){
                num = i;
                backtrack(num, low, high, result);
                num = 0;
            }
        }
        else{
            int pre = num % 10;
            if (pre < 9){
                num = num * 10 + (pre + 1);
                backtrack(num, low, high, result);
                num /= 10;
            }
        }
    }
    
    vector<int> sequentialDigits(int low, int high) {
        vector<int> result;
        long long num = 0;
        
        backtrack(num, low, high, result);
        
        sort(result.begin(), result.end());
        
        return result;
    }
};